python-spec/src/somacore/images.py

```diff
@@ -0,0 +1,81 @@
+"""Implementation of the SOMA image collection for spatial data"""
+
+import abc
+from typing import Generic, Optional, Tuple, TypeVar
+
+import pyarrow as pa
+from typing_extensions import Final, Protocol
+
+from . import base
+from . import collection
+from . import coordinates
+from . import data
+from . import options
+
+_DenseND = TypeVar("_DenseND", bound=data.DenseNDArray)
+"""A particular implementation of a collection of DenseNDArrays."""
+_RootSO = TypeVar("_RootSO", bound=base.SOMAObject)
+"""The root SomaObject type of the implementation."""
+
+_RO_AUTO = options.ResultOrder.AUTO
+
+
+class Image2D(
+    collection.BaseCollection[_RootSO],
+    Generic[_DenseND, _RootSO],
+):
```

Comment:
I created an image collection for 2D images only. This is slightly different from early high-level discussions. I am not adding a 3D image collection now, but we can add one in the future.

Reply:
I think we should name this `Image2DCollection` to make it clear it is one, otherwise it can be confused to and actual image array. It also follow the general naming structure of other SOMA objects, e.g. Sparse**Array** and Dense**Array**